src/home_assistant.rs

```diff
@@ -60,6 +63,20 @@ pub struct Number {
     unit_of_measurement: String,
 }
 
+// https://www.home-assistant.io/integrations/text.mqtt/
+#[derive(Debug, Serialize)]
+pub struct Text {
```

Comment:
https://github.com/home-assistant/core/blob/master/homeassistant/components/mqtt/discovery.py#L46 shows us all the available controls; just to confirm that there isn't a time, much less a "time range with minute resolution".